packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue

```diff
@@ -110,8 +119,75 @@ export default defineComponent({
       'sharesTreeLoading'
     ]),
     ...mapGetters(['capabilities', 'configuration']),
-    ...mapState('Files', ['sharesTree']),
     ...mapState(['user']),
+    ...mapState('Files', ['sharesTree']),
+
+    addButtonLabel() {
+      return this.$gettext('Add link')
+    },
+
+    defaultNewLinkName() {
+      return this.capabilities?.files_sharing?.public?.defaultPublicLinkShareName || ''
+    },
+
+    globalExpirationDate() {
+      const expireDate = this.capabilities.files_sharing.public.expire_date
+
+      let defaultExpireDate = null
+      let maxExpireDateFromCaps = null
+
+      if (expireDate.days) {
+        const days = parseInt(expireDate.days)
+        defaultExpireDate = DateTime.now()
+          .setLocale(this.$language.current)
+          .plus({ days })
+          .toJSDate()
+      }
+
+      if (expireDate.enforced) {
+        const days = parseInt(expireDate.days)
+        maxExpireDateFromCaps = DateTime.now()
+          .setLocale(this.$language.current)
+          .plus({ days })
+          .toJSDate()
+      }
+
+      return {
+        enforced: expireDate.enforced,
+        default: defaultExpireDate,
+        min: DateTime.now().setLocale(this.$language.current).toJSDate(),
+        max: maxExpireDateFromCaps
+      }
+    },
+
+    availableRoleOptions() {
+      const roles = LinkShareRoles.list(
+        this.highlightedFile.isFolder,
+        this.capabilities.files_sharing?.public?.can_edit
+      ).map((role) => {
+        return {
+          role,
+          name: role.name,
+          label: this.$gettext(role.label)
+        }
+      })
+
+      // add empty permission link if oCIS for alias link
+      return [
+        // { role: null, name: 'Alias link', label: this.$gettext('Only invited people') },
```

Comment:
leftover?

Reply:
no, reminder for future implementation (alias link / permissions=0), but I'd a prefer an actual ShareRole instance here instead of JSON.

Reply:
To be cleaned up in the near future in another PR